.github/workflows/linting.yml

```diff
@@ -35,3 +35,25 @@ jobs:
       - name: Ruff format
         if: ${{ always() }}
         run: ruff format . --check
+
+      - name: cmake-format
+        if: ${{ always() }}
+        run: |
+          shopt -s globstar
+          cmake-format --check **/CMakeLists.txt
+
+      - name: clang-format-diff
+        if: ${{ always() }}
+        run: |
+          git fetch origin main:main
+          export MAIN_SHA=`cat .git/refs/heads/main`  # Works for non-PR runs
+          echo MAIN_SHA=${MAIN_SHA}
+          CLANG_FORMAT_DIFF=`git diff -U0 ${MAIN_SHA} -- '*.h' '*.c' '*.hpp' '*.cpp' | clang-format-diff.py -p1`
```

Comment:
Why only look at the diff?

Reply:
Good point, this was unnecessarily complex. Adjusted to a more sane way of doing it